ocfweb/lab_reservations/reserve.py

```diff
@@ -0,0 +1,178 @@
+    :param endtime:
+    """
+
+    def to_dict(self):
+        return {
+            field: getattr(self, field)
+            for field in self._fields
+        }
+
+
+class RequestForm(forms.Form):
+
+    error_css_class = 'error'
+    required_css_class = 'required'
+
+    real_name = forms.CharField(
+        label='Full Name',
+        widget=forms.TextInput(attrs={'placeholder': 'Jane Doe'}),
+        min_length=3,
+        max_length=32,
+    )
+
+    contact_email = forms.EmailField(
+        label='Contact e-mail',
+        validators=[wrap_validator(ocflib.misc.validators.valid_email)],
+        widget=forms.EmailInput(attrs={'placeholder': 'janedoe@berkeley.edu'}),
+    )
+
+    verify_contact_email = forms.EmailField(
+        label='Confirm contact e-mail',
+        widget=forms.EmailInput(attrs={'placeholder': 'janedoe@berkeley.edu'}),
+    )
+
+    student_group = forms.CharField(
+        label='Student Group',
+        widget=forms.TextInput(attrs={'placeholder': 'OCF'}),
+        min_length=3,
+        max_length=32,
+    )
+
+    reason = forms.CharField(
+        label='Reason for reservation',
+        widget=forms.TextInput(attrs={'placeholder': ''}),
+    )
+
+    date = forms.DateField(
+        label='Date of reservation (mm/dd/yy)',
+        widget=forms.DateInput(attrs={'placeholder': '02/28/18'}),
+    )
+
+    starttime = forms.TimeField(
+        label='Starting time of reservation (xx:xx)',
+    )
+
+    endtime = forms.TimeField(
+        label='Ending time of reservation (xx:xx)',
+    )
+
+    disclaimer_agreement = forms.BooleanField(
+        label='I agree with the above statement.',
+        error_messages={
+            'required': 'You must agree to our policies.',
+        },
+    )
+
+    def clean_verify_contact_email(self):
+        email = self.cleaned_data.get('contact_email')
+        verify_contact_email = self.cleaned_data.get('verify_contact_email')
+
+        if email and verify_contact_email:
+            if email != verify_contact_email:
+                raise forms.ValidationError("Your emails don't match.")
+        return verify_contact_email
+
+
+@calnet_required
+def request_reservation(request):
+    calnet_uid = request.session['calnet_uid']
+    status = 'new_request'
+
+    # ensure we can even find them in university LDAP
+    # (alumni etc. might not be readable in LDAP but can still auth via CalNet)
+    if not user_attrs_ucb(calnet_uid):
+        return render(
+            request,
+            'lab_reservations/lab_reservations/cant-find-in-ldap.html',
+            {
+                'calnet_uid': calnet_uid,
+                'title': 'Unable to read account information',
+            },
+        )
+
+    if request.method == 'POST':
+        form = RequestForm(request.POST)
+        if form.is_valid():
+            req = NewReservationRequest(
+                real_name=form.cleaned_data['real_name'],
+                contact_email=form.cleaned_data['contact_email'],
```

Comment:
Is there anything that validates this email address? We have a function in ocflib which is good at this (it even checks the DNS name). You wouldn't believe how many typos of `berkeley.edu` it's caught, even from Berkeley students.

Reply:
Within the RequestForm class, we have a line `validators=[wrap_validator(ocflib.misc.validators.valid_email)],`

Does this assuage any concerns?

Reply:
yeah, i must have missed that. assuming it works, i think this is fine.